lib/pavilion/builder.py

```diff
@@ -586,12 +592,26 @@ def _build(self, build_dir, cancel_event):
                                         stderr=build_log)
 
                 result = None
+                start = time.time()
+                deleted = None
+                file_created = False
                 while result is None:
                     try:
                         result = proc.wait(timeout=1)
                     except subprocess.TimeoutExpired:
-                        log_stat = self.tmp_log_path.stat()
-                        timeout = log_stat.st_mtime + self._timeout
+                        try:
+                            log_stat = self._timeout_file.stat()
+                            file_created = True
+                            timeout = log_stat.st_mtime + self._timeout
+                        except FileNotFoundError as err:
+                            # Timeout file has not been created yet.
+                            if not file_created:
+                                timeout = start + self._timeout
+                            # Timeout file existed, but was deleted.
+                            else:
+                                if deleted is None:
+                                    deleted = time.time()
+                                timeout = deleted + self._timeout
```

Comment:
Let's simplify the problem a bit. 

1. If the timeout file doesn't exist, use the log file.
2. `timeout = max(timeout, stat.st_mtime + self._timeout)`

Then you don't have to track creation or deletion, and your only conditional is `if self._timeout_file.exists(): timeout_file = ...`

Reply:
I tweaked your fix to this a little bit, and addressed a race condition in the process.